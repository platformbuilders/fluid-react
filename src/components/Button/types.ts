import { ReactNode } from 'react';
import {
  ButtonColorType,
  ButtonVariants,
  TypographyVariants,
} from '@platformbuilders/theme-toolkit';
import RadixIcon from '@radix-ui/react-icons';

export type ButtonProps = {
  style?: any;
  type?: 'button' | 'submit' | 'reset';
  loading?: boolean;
  $fullWidth?: boolean;
  variant?: ButtonVariants;
  typographyVariant?: TypographyVariants;
  children?: string | ReactNode;
  leftIconName?: keyof typeof RadixIcon;
  rightIconName?: keyof typeof RadixIcon;
  colorVariant?: ButtonColorType;
  size?: 'normal' | 'medium' | 'large';
} & TouchableType;

export type TouchableType = {
  id?: string;
  accessibility: string;
  testID?: string;
  disabled?: boolean;
  onPress?: (param: any) => void;
};
