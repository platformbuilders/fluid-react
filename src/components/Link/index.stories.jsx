import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import markdown from './if.md';
import Link from './index';

storiesOf('Link', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: markdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => <Link onPress={() => { }}>Click Here</Link>);
