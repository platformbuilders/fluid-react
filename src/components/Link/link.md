
# Link

## Link component with similar functionality to anchor element

```js
import React from 'react';
import { Link } from '@platformbuilders/react-ui';

const Home = () => {
   <Link onPress={() => {}}>
      Click here
   </Link>
  );
}

export default Home;
```


| Name  | Required | Type | Default Value | Description
| ------------- | ------------- | ------------- |------------- |------------- |
| condition | true | boolean |  |  |
