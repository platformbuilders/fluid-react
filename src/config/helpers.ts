import { DefaultTheme } from 'styled-components';
import { animations, colors, radius, spacing } from '../theme';

export const theme: DefaultTheme = {
  ...colors,
  ...animations,
  ...radius,
  ...spacing,
};
