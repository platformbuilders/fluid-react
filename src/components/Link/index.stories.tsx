import type { Meta, StoryObj } from '@storybook/react';

import Link from './index';

const meta: Meta<typeof Link> = {
  title: 'Link',
  component: Link,
  args: {
    children: 'Text to link',
  },
};

export default meta;
type Story = StoryObj<typeof Link>;

export const Default: Story = {};
