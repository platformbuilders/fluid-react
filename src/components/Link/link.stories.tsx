import { expect, jest } from '@storybook/jest';
import type { Meta, StoryObj } from '@storybook/react';
import { within } from '@storybook/testing-library';
import Link from './index';

//  Mocks
const mockTxt = 'Link Test';
const mockFn = jest.fn();

const meta: Meta<typeof Link> = {
  title: 'Components/Link',
  component: Link,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    children: {
      description: 'The content of the link',
      control: {
        type: 'text',
      },
    },
    onPress: {
      description: 'The function to be called when the link is clicked',
      action: 'clicked',
      control: {
        type: null,
      },
    },
  },
  args: {
    children: mockTxt,
    onPress: mockFn,
  },
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Link | Test Render', async () => {
      expect(canvas.getByText(mockTxt)).toBeInTheDocument();
    });

    await step('Link | Test Click', async () => {
      await canvas.getByText(mockTxt).click();
      expect(mockFn).toHaveBeenCalled();
    });
  },
};

type Story = StoryObj<typeof Link>;

export const Default: Story = {
  args: {},
};

export default meta;
