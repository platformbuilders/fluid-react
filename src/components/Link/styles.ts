import styled from 'styled-components';
import { Link } from '@material-ui/core';
import { getTheme } from '../../utils/helpers';

const primary = getTheme('primary.main');

export const CustomLink = styled(Link)`
  color: ${primary};
  text-decoration: none;
  cursor: pointer;
  &:hover {
    text-decoration: none;
  }
`;
