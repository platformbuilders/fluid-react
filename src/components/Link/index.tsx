import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { Touchable } from '~/components';
import { Variant } from '~/utils/types';
import { Text } from './styles';
import { ThemeContext } from '../ThemeContext';

type Props = {
  children: string;
  onPress: () => null;
  variant?: Variant;
  accessibility: string;
};

const Link: React.FC<Props> = ({
  onPress,
  children,
  accessibility,
  variant = 'body',
  ...rest
}) => {
  const { theme } = useContext(ThemeContext);
  return (
    <ThemeProvider theme={theme}>
      <Touchable onPress={onPress} accessibility={accessibility} {...rest}>
        <Text variant={variant}>{children}</Text>
      </Touchable>
    </ThemeProvider>
  );
};

export default Link;
