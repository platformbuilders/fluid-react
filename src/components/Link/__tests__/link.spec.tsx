import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import Link from '..';
import { theme } from '../../../config/helpers';
import { LinkProps as Props } from '../../../types/Link';

const defaultContent = 'Text';
const defaultProps: Props = {
  onPress: jest.fn(),
  children: defaultContent,
};

describe('Component: Link', () => {
  test('snapshots with default props', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <Link {...defaultProps} />
      </ThemeProvider>,
    );

    expect(component).toMatchSnapshot();
  });

  test('should call onPress when pressed', () => {
    // should
    const mockFunction = jest.fn();
    const component = shallow(
      <Link {...defaultProps} onPress={mockFunction}>
        {defaultContent}
      </Link>,
    );

    // when
    expect(mockFunction).not.toHaveBeenCalled();
    // component.props().onClick();
    component.simulate('click');

    // then
    expect(mockFunction).toHaveBeenCalled();
  });
});
