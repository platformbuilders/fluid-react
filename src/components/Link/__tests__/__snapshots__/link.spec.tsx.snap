// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Link snapshots with default props 1`] = `
<ContextProvider
  value={Object {}}
>
  <Link
    onPress={[MockFunction]}
  >
    Text
  </Link>
</ContextProvider>
`;
